src/ripple/protocol/TxFlags.h

```diff
@@ -159,6 +160,9 @@ constexpr std::uint32_t const tfNFTokenCancelOfferMask     = ~(tfUniversal);
 // NFTokenAcceptOffer flags:
 constexpr std::uint32_t const tfNFTokenAcceptOfferMask     = ~tfUniversal;
 
+// Clawback flags:
+constexpr std::uint32_t const tfClawbackMask     = ~tfUniversal;
```

Comment:
I did not understand the intent of this variable. I know that this variable is used inside `preflight()`. Source file: `src/ripple/app/tx/impl/Clawback.cpp`

If any of the universal flags are set, then return `temINVALID_FLAG`, is this correct?

Reply:
It does the opposite, if any flags other than the universal flag are set, then return `temINVALID_FLAG`.

Here, clawback does not introduce any new flags

In binary:
tfUniversal = `10000000000000000000000000000000` 
tfClawbackMask =  `01111111111111111111111111111111`

If someone enters a flag into the `Clawback` transaction such as `00000000000000000000000000000001`, then
ctx.tx.getFlags() & tfClawbackMask = `00000000000000000000000000000001 & 01111111111111111111111111111111 ` will result in `00000000000000000000000000000001`, which will return `temINVALID_FLAG`

Therefore, we know the flag value of 1 is not valid.

Reply:
@scottschurr just curious, is there any use for the `tfFullyCanonicalSig` flag? i think it's deprecated

Reply:
Hi @shawnxie999.  The `tfFullyCanonicalSig` flagis not deprecated, however it has had no effect since July of 2020.  On July 3, 2020 an amendment went live that requires all signatures to be fully canonical, regardless of whether the `tfFullyCanonicalSig ` is set.  So it's okay to set the flag, it just doesn't make things any safer than otherwise.

We certainly don't want the server to reject any transactions with the `tfFullyCanonicalSig ` flag set.  So the flag itself is still valid.

Reply:
Thank you!

Reply:
ok, thanks for the detailed explanation @shawnxie999

Reply:
Why are we not allowing any of the non-universal flags along with a `Clawback` transaction? Where can I learn more about universal flags and their uses?

Reply:
@ckeshava `Clawback` here does not have introduce any transaction-specific flag, there it does not have any non-universal flags.

The only universal flag that is used by every transaction seems to be  `tfFullyCanonicalSig`, as defined here in the [code](https://github.com/XRPLF/rippled/blob/77c0a62a7472dbc4dc9ed97d3141840a3acb5ce8/src/ripple/protocol/TxFlags.h#L59).

Reply:
ok, thank you!